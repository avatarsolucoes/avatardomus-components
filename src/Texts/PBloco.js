/* eslint-disable react/prop-types */
import React from 'react';
import cx from 'classnames';

import css from './texts.css';

export default function PBloco(props) {
  const { children, className, right, line, ...rest } = props;
  const styles = right ? { textAlign: 'right', ...rest.style } : { ...rest.style };

  const classe = cx(css.pBloco, { [css.line]: !!line }, 'pBloco', { line: !!line }, className);
  return (
    <p className={classe} {...rest} style={styles}>
      {children}
    </p>
  );
}
