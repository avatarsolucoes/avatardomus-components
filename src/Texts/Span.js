import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import css from './texts.css';

export default function Span(props) {
  const { children, text, className, size, bold, strong, ...rest } = props;
  const styles = size ? { fontSize: size, ...rest.style } : { ...rest.style };

  const classCss = {
    [css.textSpan]: true,
    [css.bold]: !!bold,
  };

  const classe = cx(classCss, 'text-span', { bold: !!bold }, className);
  return (
    <>
      {strong ? (
        <strong className={classe} {...rest} style={styles}>
          {children || text}
        </strong>
      ) : (
        <span className={classe} {...rest} style={styles}>
          {children || text}
        </span>
      )}
    </>
  );
}

Span.propTypes = {
  children: PropTypes.any,
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  className: PropTypes.string,
  size: PropTypes.number,
  bold: PropTypes.bool,
  strong: PropTypes.bool,
};

Span.defaultProps = {
  children: null,
  text: null,
  className: null,
  size: null,
  bold: false,
  strong: false,
};
