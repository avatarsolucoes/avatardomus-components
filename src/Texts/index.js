export { default as Nota } from './Nota';
export { default as Span } from './Span';
export { default as TextOn } from './TextOn';
export { default as TextOff } from './TextOff';
export { default as TextOnOff } from './TextOnOff';
export { default as Text } from './Text';
export { default as PBloco } from './PBloco';
