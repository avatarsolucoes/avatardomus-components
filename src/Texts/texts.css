.textSpan {
  padding: 0;
  margin: 0;
}

.textOn {
  color: #6ab16a;
  padding: 0;
}

.textOff {
  color: rgba(255, 255, 255, 0.7);
}

.pBloco {
  display: block;
  max-width: 100%;
}

.pBloco.line {
  border-bottom: rgba(255, 255, 255, 0.7) 1px solid;
}

.text-suplabel {
  position: relative;
  display: block;
  margin: 0;
  padding: 0;
  font-size: 10px;
  line-height: 10px;
  opacity: 0.7;
}

.textNota {
  position: relative;
  margin: 0;
  padding: 0;
  font-size: 10px;
  display: block;
}

.bold {
  font-weight: bold;
}

.weak {
  opacity: 0.6;
}

.f14{ font-size: 14px; }

.pad2{ padding: 2px; }
.pad4{ padding: 4px; }
.pad6{ padding: 6px; }
.pad8{ padding: 8px; }
.pad10{ padding: 10px; }

.padH0{ padding-left: 0; padding-right: 0; }
.padV0{ padding-top: 0; padding-bottom: 0;}


.pV0L4{ padding: 0 4px; }


