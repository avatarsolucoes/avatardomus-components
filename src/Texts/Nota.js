import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import css from './texts.css';

import Span from './Span';

export default function Nota(props) {
  const { children, text, bold, inline, className, style, ...rest } = props;

  const classCss = {
    [css.textNota]: true,
    [css.bold]: !!bold,
  };

  const classe = cx(classCss, 'text-nota', { bold: !!bold }, className);

  const styles = { ...style };
  if (inline) styles.display = 'inline';

  return (
    <Span style={styles} className={classe} {...rest}>
      {children || text}
    </Span>
  );
}

Nota.propTypes = {
  children: PropTypes.any,
  text: PropTypes.string,
  className: PropTypes.string,
  size: PropTypes.number,
  bold: PropTypes.bool,
  strong: PropTypes.bool,
  inline: PropTypes.bool,
  style: PropTypes.object,
};

Nota.defaultProps = {
  children: null,
  text: null,
  className: null,
  size: null,
  bold: false,
  strong: false,
  inline: null,
  style: {},
};
