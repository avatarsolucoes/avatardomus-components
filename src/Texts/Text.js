import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import css from './texts.css';

import Span from './Span';

export default function Text(props) {
  const { children, text, className, onDoubleClick, onClick, bold, ...rest } = props;

  const classCss = {
    [css.f14]: true,
  };

  const classe = cx(classCss, className);

  return (
    <Span
      bold={!!bold}
      className={classe}
      onDoubleClick={onDoubleClick}
      onClick={onClick}
      {...rest}
    >
      {children || text}
    </Span>
  );
}

Text.propTypes = {
  children: PropTypes.any,
  text: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  className: PropTypes.string,
  onDoubleClick: PropTypes.func,
  onClick: PropTypes.func,
  bold: PropTypes.bool,
  title: PropTypes.string,
  style: PropTypes.object,
};

Text.defaultProps = {
  children: null,
  text: null,
  className: '',
  onDoubleClick: () => {},
  onClick: () => {},
  bold: false,
  title: '',
  style: {},
};
