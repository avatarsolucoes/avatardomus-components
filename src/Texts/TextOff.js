/* eslint-disable react/prop-types */
import React from 'react';
import cx from 'classnames';
import css from './texts.css';

import Span from './Span';

export default function TextOff(props) {
  const { children, text, className, ...rest } = props;
  const classe = cx(css.TextOff, 'text-off', 'contrast', className);
  return (
    <Span className={classe} {...rest}>
      {children || text}
    </Span>
  );
}
