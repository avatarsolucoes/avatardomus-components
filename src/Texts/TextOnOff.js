/* eslint-disable react/prop-types */
import React from 'react';
import PropTypes from 'prop-types';

import TextOn from './TextOn';
import TextOff from './TextOff';

export default function TextOnOff(props) {
  const { children, text, className, online, ...rest } = props;
  const Result = online ? TextOn : TextOff;
  const defaultText = online ? 'Online' : 'Offline';

  return (
    <Result className={className} {...rest}>
      {children || text || defaultText}
    </Result>
  );
}
TextOnOff.propTypes = {
  online: PropTypes.bool,
  text: PropTypes.string,
};
TextOnOff.defaultProps = {
  online: false,
  text: null,
};
