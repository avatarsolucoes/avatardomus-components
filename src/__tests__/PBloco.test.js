import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from '../../setupEnzyme';

import { PBloco } from '../Texts';

describe('PBloco', () => {
  it(`is truthy PBloco`, () => {
    expect(PBloco).toBeTruthy();
  });

  it(`Test PBloco`, () => {
    const wrapper = {
      line: renderer.create(<PBloco line>CHILDREN</PBloco>).toJSON(),
      bloco: renderer.create(<PBloco right>CHILDREN</PBloco>).toJSON(),
    };
    expect(wrapper).toMatchSnapshot();
  });
});
