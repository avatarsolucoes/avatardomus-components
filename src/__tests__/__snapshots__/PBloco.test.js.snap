// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PBloco Test PBloco 1`] = `
Object {
  "bloco": <p
    className="pBloco"
    style={
      Object {
        "textAlign": "right",
      }
    }
  >
    CHILDREN
  </p>,
  "line": <p
    className="undefined pBloco line"
    style={Object {}}
  >
    CHILDREN
  </p>,
}
`;
